frontend/src/services/UtilsService.ts

```diff
@@ -60,12 +64,54 @@ function getLargestHeader(headers: Array<string>, data?: Array<any>) {
   );
 }
 
+const defaultColors = [
+  "#2491ff",
+  "#54278f",
+  "#c05600",
+  "#002d3f",
+  "#00a398",
+  "#c2850c",
+  "#fd4496",
+  "#3e4ded",
+  "#008817",
+  "#5c1111",
+  "#e52207",
+  "#ab2165",
+  "#0f6460",
+];
+
+const getColors = (
+  numberOfColors: number,
+  primaryColor: string,
+  secondaryColor: string
+): Array<string> => {
+  const colors = new Array<string>();
+  colors.push(primaryColor);
+  const secondaryIndex = defaultColors.indexOf(secondaryColor);
+  for (let i = secondaryIndex; i < numberOfColors + secondaryIndex; i++) {
+    const color = defaultColors[i % defaultColors.length];
+    if (color !== primaryColor) {
+      colors.push(color);
+    }
+  }
+  return colors;
+};
+
+const getSecondaryOptions = () => {
+  return defaultColors.map((c) => {
+    return { value: c, content: c };
+  });
+};
+
```

Comment:
We need to be cautious about our UtilsService growing too much. It's very tempting to just throw functions into Utils but it shouldn't be our default place to put them. In this case, the colors were already part of a `useColors` hook which was a good place to be. I don't think the UtilsService is the right place for them.

Reply:
In my opinion, Utils should be very specific functions that perform "plumbing" work that is not really part of our Domain or business logic. Example: Functions to manipulate an array or a string. In this case, the Color Palette is something important in our core domain and I think it deserves it's own service `ColorPaletteService` or to put it back into a `useColors` hook. 

What do you think?

Reply:
Agree, I'll commit the updates shortly, I did not like it the useColors hooks the way it existed but with the Color palette service now it makes sense.